cli/util/sync.rs

```diff
@@ -19,9 +25,147 @@ impl AtomicFlag {
   }
 }
 
+#[derive(Debug, Default)]
+struct TaskQueueTaskItem {
+  is_ready: AtomicFlag,
+  is_future_dropped: AtomicFlag,
+  waker: AtomicWaker,
+}
+
+#[derive(Debug, Default)]
+struct TaskQueueTasks {
+  is_running: bool,
+  items: LinkedList<Arc<TaskQueueTaskItem>>,
+}
+
+/// A queue that executes tasks sequentially one after the other
+/// ensuring order and that no task runs at the same time as another.
+///
+/// Note that this differs from tokio's semaphore in that the order
+/// is acquired synchronously.
```

Comment:
I've made some changes to this so that the order is acquired synchronously similar to TaskQueue in deno_unsync. I need this functionality for the semvered specifier tests.